indra/newview/llfloaterimsessiontab.cpp

```diff
@@ -433,16 +434,34 @@ void LLFloaterIMSessionTab::draw()
 
 void LLFloaterIMSessionTab::enableDisableCallBtn()
 {
-    if (LLVoiceClient::instanceExists() && mVoiceButton)
+    if (!mVoiceButton)
+        return;
+
+    BOOL enable = FALSE;
+
+    if (mSessionID.notNull() && mSession && mSession->mSessionInitialized && mSession->mCallBackEnabled)
     {
-        mVoiceButton->setEnabled(
-            mSessionID.notNull()
-            && mSession
-            && mSession->mSessionInitialized
-            && LLVoiceClient::getInstance()->voiceEnabled()
-            && LLVoiceClient::getInstance()->isVoiceWorking()
-            && mSession->mCallBackEnabled);
+        if (mVoiceButtonHangUpMode)
+        {
+            // We allow to hang up from any state
+            enable = TRUE;
```

Comment:
Why isn't this checking voice state to make sure there is something to hung up? Like an outgoing attempt of a call.

Reply:
> Why isn't this checking voice state to make sure there is something to hung up? Like an outgoing attempt of a call.

See the comment below https://github.com/secondlife/viewer/pull/2381#issuecomment-2304226886